lib/gtfs/data.ex

```diff
@@ -68,6 +82,54 @@ defmodule Gtfs.Data do
     Block.get(blocks, block_id, service_id)
   end
 
+  @spec active_blocks(t(), Util.Time.timestamp(), Util.Time.timestamp()) :: %{
+          Route.id() => [Block.id()]
+        }
+  def active_blocks(%__MODULE__{trips: trips, calendar: calendar}, start_time, end_time) do
+    # We might cover multiple service dates
+    # If we start before 6am, check trips on the previous day
+    # It's okay if this date range is a little too broad, since we check trips by time on those dates later
+    first_possible_service_date =
+      start_time
+      |> (fn time -> time - 6 * 60 * 60 end).()
+      |> Util.Time.date_of_timestamp()
+
+    last_possible_service_date =
+      end_time
+      |> Util.Time.date_of_timestamp()
+
+    date_range = Date.range(first_possible_service_date, last_possible_service_date)
+    dates = Enum.to_list(date_range)
+    active_services = Map.take(calendar, dates)
+
+    active_trips =
+      Enum.flat_map(active_services, fn {date, service_ids} ->
```

Comment:
It's not necessary, but I could see this getting refactored out as a function on `Trip`.

Reply:
Hmm.. I just tried but the service / time range comparisons are kind of specific to what's going on in this file so it didn't really feel like an improvement.